lib/Target/ARM/ARMRelocator.cpp

```diff
@@ -1082,7 +1069,7 @@ Relocator::Result thm_jump11(Relocation &pReloc, ARMRelocator &pParent) {
             ->getAddr(DiagEngine);
 
   Relocator::DWord X = S + A - P;
-  if (helper_check_signed_overflow(X, 12))
+  if (!llvm::isInt<12>(static_cast<int32_t>(X)))
```

Comment:
Why are we casting it all to int32_t? Is there any issue if we keep them 64-bits?